certif/tests/unit/services/session_test.js

```diff
@@ -0,0 +1,197 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import sinon from 'sinon';
+
+module('Unit | Service | session', function (hooks) {
+  setupTest(hooks);
+
+  let sessionService;
+  let routerService;
+
+  hooks.beforeEach(function () {
+    sessionService = this.owner.lookup('service:session');
+    sessionService.currentDomain = { getExtension: sinon.stub() };
+    sessionService.currentUser = { load: sinon.stub(), certificationPointOfContact: null };
+
+    routerService = this.owner.lookup('service:router');
+    routerService.transitionTo = sinon.stub();
+  });
+
+  module('#handleAuthentication', function () {
+    test('calls loadCurrentUserAndSetLocale with no arg', async function (assert) {
+      // given
+      sessionService.loadCurrentUserAndSetLocale = sinon.stub();
+      sessionService.currentUser = { certificationPointOfContact: { isMemberOfACertificationCenter: false } };
+      sessionService._setLocale = sinon.stub();
+
+      // when
+      await sessionService.handleAuthentication();
+
+      // then
+      sinon.assert.calledWith(sessionService.loadCurrentUserAndSetLocale);
+      assert.ok(true);
+    });
+  });
+
+  module('#loadCurrentUserAndSetLocale', function () {
+    module('when domain is .fr', function () {
+      module('when no lang query parameter', function () {
+        module('when user is not loaded', function () {
+          test('calls _setLocale with fr', async function (assert) {
+            // given
+            sessionService.currentDomain.getExtension.returns('fr');
+            sessionService._setLocale = sinon.stub();
+
+            // when
+            await sessionService.loadCurrentUserAndSetLocale();
+
+            // then
+            sinon.assert.calledWith(sessionService._setLocale, 'fr');
+            assert.ok(true);
+          });
+        });
+
+        module('when user is loaded', function () {
+          test('calls _setLocale with fr', async function (assert) {
+            // given
+            sessionService.currentDomain.getExtension.returns('fr');
+            sessionService.currentUser.certificationPointOfContact = {
+              lang: 'user lang has no impact in this scenario',
+            };
+            sessionService._setLocale = sinon.stub();
+
+            // when
+            await sessionService.loadCurrentUserAndSetLocale();
+
+            // then
+            sinon.assert.calledWith(sessionService._setLocale, 'fr');
+            assert.ok(true);
+          });
+        });
+      });
+      module('when lang query parameter', function () {
+        module('when user is not loaded', function () {
```

Comment:
Remarque : on dirait que la condition première est le fait que le user est pas loadée alors qu'on est dans le cas où on a un domaine en .fr et on s'en fiche des query params, c'est pas hyper clair je trouve.
Suggestion : 
- module('when lang in query parameters'
- module('but domain is .fr'
- module('no matter if user is loaded or not' 
- test('sets locals with FRENCH_INTERNATIONAL_LOCALE'

Reply:
Les intitulés des tests ont été modifiés pour être « métier » et non plus « technique ».